lib/interfaces/gql-gateway-module-options.interface.ts

```diff
@@ -0,0 +1,30 @@
+import { Type } from '@nestjs/common';
+import { GqlModuleOptions } from './gql-module-options.interface';
+import { GatewayConfig, ServiceEndpointDefinition } from '@apollo/gateway';
+import { GraphQLDataSource } from '@apollo/gateway/src/datasources/types';
+import { ModuleMetadata } from '@nestjs/common/interfaces';
+
+export interface GatewayModuleOptions {
+  gateway?: GatewayConfig;
+  server?: Pick<
```

Comment:
@tuxmachine you pick here only few fields that we can pass to apollo when we create our server. 
Is there any particular reason for that? 😕 

This prevents us, for example, from initializing and configuring Apollo Graph Manager with the field `engine`, which can be quite annoying for some implementations with a distributed graph.

Reply:
Again, I followed the original code of @marcus-sa that did a `Pick<GqlModuleOptions>`, maybe he can chime in with his original intention? I assume it's because the code doesn't support all the options, nor do they make sense for the gateway (e.g. `autoSchemaFile`) 

Are there any other fields you are missing?

Reply:
I inverted the Pick so the engine prop becomes available